src/util.h

```diff
@@ -53,6 +53,13 @@ namespace node {
 
 #define UNREACHABLE() ABORT()
 
+#define ASSIGN_OR_RETURN_UNWRAP(TYPE, PTR, OBJ, ...)                          \
+  do {                                                                        \
+    PTR = Unwrap<TYPE>(OBJ);                                                  \
```

Comment:
I don't think you need to pass the type explicitly, just `PTR = Unwrap<decltype(*PTR)>(OBJ)` should suffice.

Aside, using all caps for macro arguments is a bit off-kilter.